src/main/java/org/elasticsearch/action/get/TransportGetAction.java

```diff
@@ -74,6 +75,14 @@ protected void resolveRequest(ClusterState state, InternalRequest request) {
         if (request.request().realtime == null) {
             request.request().realtime = this.realtime;
         }
+        IndexMetaData indexMeta = state.getMetaData().index(request.concreteIndex());
+        if (request.request().realtime && // if the realtime flag is set
+                request.request().preference() == null && // the preference flag is not already set
+                indexMeta != null && // and we have the index
+                IndexMetaData.isIndexUsingShadowReplicas(indexMeta.settings())) { // and the index uses shadow replicas
+            // set the preference for the request to use "_primary" automatically
+            request.request().preference("_primary");
```

Comment:
can we use `org.elasticsearch.cluster.routing.operation.plain.Preference.PRIMARY.type()` here instead?